### PR TITLE
Hide reviews from abstract PDF export depending on role

### PR DESCRIPTION
**Pull request by mvidalgarcia**
Adds a dropdown menu for users that can see the reviews to choose whether they want to export the PDF with or without reviews.

- [x] Fix bug when deleting comments in abstract review

### CODE REVIEW

File: indico/modules/events/abstracts/templates/abstract/public.html [hunk 1]
```diff
@@ -301,12 +301,30 @@ <h3 class="abstract-title">
...
+                    <a class="i-button arrow icon-file-download dropdown-abstract-pdf"
+                       data-toggle="dropdown"></a>
+                    <ul class="dropdown">
+                        <li>
+                            <a href="{{ url_for('.display_abstract_pdf_export', abstract) }}"
+                               class="icon-file-pdf">PDF</a>
+                        </li>
+                        <li>
+                            <a href="{{ url_for('.manage_abstract_pdf_export', abstract) }}"
+                               class="icon-file-pdf">{% trans %}PDF with reviews{% endtrans %}</a>
```
Comment (mvidalgarcia) [inline indico/modules/events/abstracts/templates/abstract/public.html]: Not pretty sure about these names and whether they should have the PDF icon.

Comment (ThiefMaster) [inline indico/modules/events/abstracts/templates/abstract/public.html]: Maybe having no icons on the dropdown items would be the best option? All options are for PDF files anyway.

File: indico/htdocs/js/indico/modules/abstracts.js [hunk 2]
```diff
@@ -39,6 +39,7 @@
     };
 
     global.setupAbstractPage = function setupAbstractPage() {
+        $('.dropdown-abstract-pdf').parent().dropdown();
```
Comment (mvidalgarcia) [inline indico/htdocs/js/indico/modules/abstracts.js]: I haven't thought about that. I've just checked again and the dropdown stops working when e.g somebody leaves a comment. Do you know how can I solve it?

File: indico/modules/events/abstracts/controllers/management.py [hunk 3]
```diff
@@ -154,6 +154,10 @@ def _process(self):
 
 
 class RHAbstractExportPDF(RHManageAbstractBase):
+    def _checkProtection(self):
```
Comment (OmeGak) [inline indico/modules/events/abstracts/controllers/management.py]: I don't think we need to add this check here. We are extending rom `RHManageAbstractBase`. Doesn't that checks only managers have access? Why wouldn't a manager be able to see reviews?

File: indico/modules/events/abstracts/templates/abstract/public.html [hunk 4]
```diff
@@ -479,4 +497,7 @@ <h3 class="abstract-title">
             </a>
         {% endif %}
     </div>
+    <script>
+        $('.dropdown-abstract-pdf').parent().dropdown();
```
Comment (OmeGak) [inline indico/modules/events/abstracts/templates/abstract/public.html]: I think we have some declarative way of making the dropdown elements automatically initialized.

Comment (mvidalgarcia) [inline indico/modules/events/abstracts/templates/abstract/public.html]: Yes, we have `.js-dropdown` but it doesn't work when there are AJAX reloads in the page as I mention in a comment before. If you know how to solve it please let me know.

Comment (ThiefMaster) [inline indico/modules/events/abstracts/templates/abstract/public.html]: Declarative doesn't run for ajax updates

File: indico/modules/events/abstracts/templates/abstract/public.html [hunk 5]
```diff
@@ -301,12 +301,30 @@ <h3 class="abstract-title">
                         </a>
                     {% endif %}
                 </div>
+            {% endif %}
+            {% if abstract.can_see_reviews(session.user) %}
+                <div class="group">
+                    <a class="i-button arrow icon-file-download js-dropdown-abstract-pdf"
```
Comment (ThiefMaster) [inline indico/modules/events/abstracts/templates/abstract/public.html]: I'd go for `icon-file-pdf`. We also use this in contributions for example.

**Pull request merged**